WooCommerce/WooCommerceTests/ViewRelated/CardPresentPayments/CardReaderConnectionControllerTests.swift

```diff
@@ -204,6 +204,41 @@ class CardReaderConnectionControllerTests: XCTestCase {
         wait(for: [expectation], timeout: Constants.expectationTimeout)
     }
 
+    func test_user_can_cancel_search_after_connection_error_due_to_low_battery() {
```

Comment:
It is true that all I want to do is ensure that if connecting surfaces this specific error that we trigger the specific modal. I appreciate the feedback and I think the spy idea is a good one, but I'd like to keep it similar to the other tests for the next person that comes around?

If you feel strongly about it, I will certainly open an issue :)

Reply:
> It is true that all I want to do is ensure that if connecting surfaces this specific error that we trigger the specific modal.

I don't think this test achieves that: it will still pass if you comment out the following from [`CardReaderConnectionController.showConnectionFailed(error:)`](https://github.com/woocommerce/woocommerce-ios/blob/44eb0067b0d74112b48339343314142f526e244a/WooCommerce/Classes/ViewRelated/CardPresentPayments/CardReaderConnectionController.swift#L600):

    case .bluetoothConnectionFailedBatteryCriticallyLow:
            alerts.connectingFailedCriticallyLowBattery(from: from, retrySearch: retrySearch, cancelSearch: cancelSearch)

With that removed, it will return to displaying the default modal which still satisfies the test.

With the spy, you can test directly for the appropriate modal being shown and be clear in the naming that's what the test is for.

I don't feel particularly strongly about the test, although I don't think it adds anything to the pre-existing cancellation test and may cause minor confusion for future readers, by making them think that there's something different about the cancellation flow with a critically low battery. If you don't think it's worth adding another type of test, I'd suggest just removing this one because of those two factors.